File: frontend/src/shared/types/report.ts
export interface ShiftSummary {
  shiftId: string;
  shiftName: string;
  date: string;
  startTime: string;
  endTime: string;
  unit: string;
  totalEvents: number;
  falls: number;
  bedExits: number;
  inactivityAlerts: number;
  unsafeTransfers: number;
  avgAckTimeSeconds: number;
  avgResolveTimeSeconds: number;
  unacknowledgedCount: number;
  escalatedCount: number;
  highRiskResidents: string[];
  notableIncidents: string[];
}

export interface WeeklyDigest {
  weekStart: string;
  weekEnd: string;
  unit: string;
  totalFalls: number;
  unwitnessedFalls: number;
  avgResponseTimeSeconds: number;
  medianResponseTimeSeconds: number;
  repeatFallResidents: string[];
  complianceScore: number;
  trendVsPriorWeek: 'improved' | 'unchanged' | 'declined';
}

export interface BoardReportData {
  period: string;
  facilityName: string;
  unitSummaries: {
    unit: string;
    falls: number;
    avgResponseTime: number;
    unwitnessedRate: number;
    complianceScore: number;
  }[];
  liabilityIndicators: {
    slowResponses: number;
    unacknowledgedAlerts: number;
    repeatFalls: number;
  };
  trendsMonthly: {
    month: string;
    falls: number;
    avgResponseTime: number;
  }[];
}

File: backend/src/schemas/report.py
from src.schemas.common import CamelModel


class ShiftSummaryResponse(CamelModel):
    shift_id: str
    shift_name: str
    date: str
    start_time: str
    end_time: str
    unit: str
    total_events: int
    falls: int
    bed_exits: int
    inactivity_alerts: int
    unsafe_transfers: int
    avg_ack_time_seconds: float
    avg_resolve_time_seconds: float
    unacknowledged_count: int
    escalated_count: int
    high_risk_residents: list[str]
    notable_incidents: list[str]


class WeeklyDigestResponse(CamelModel):
    week_start: str
    week_end: str
    unit: str
    total_falls: int
    unwitnessed_falls: int
    avg_response_time_seconds: float
    median_response_time_seconds: float
    repeat_fall_residents: list[str]
    compliance_score: int
    trend_vs_prior_week: str


class UnitSummary(CamelModel):
    unit: str
    falls: int
    avg_response_time: float
    unwitnessed_rate: float
    compliance_score: int


class LiabilityIndicators(CamelModel):
    slow_responses: int
    unacknowledged_alerts: int
    repeat_falls: int


class MonthlyTrend(CamelModel):
    month: str
    falls: int
    avg_response_time: float


class BoardReportResponse(CamelModel):
    period: str
    facility_name: str
    unit_summaries: list[UnitSummary]
    liability_indicators: LiabilityIndicators
    trends_monthly: list[MonthlyTrend]
